src/tests/ov_helpers/ov_snippets_models/src/subgraph_mlp.cpp

```diff
@@ -0,0 +1,319 @@
+
+#include "snippets/op/subgraph.hpp"
+
+#include "openvino/opsets/opset15.hpp"
+#include "ov_lpt_models/common/builders.hpp"
+#include "ov_lpt_models/common/dequantization_operations.hpp"
+#include "ov_lpt_models/common/fake_quantize_on_data.hpp"
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+std::shared_ptr<ov::Model> MLPSeqFunction::initOriginal() const {
+    auto A_param = std::make_shared<ov::op::v0::Parameter>(precisions[0], input_shapes[0]);
+    std::shared_ptr<Node> A = A_param;
+    if (precisions[0] != ov::element::f32) {
+        A = std::make_shared<ov::op::v0::Convert>(A, ov::element::f32);
+    }
+    auto add = std::make_shared<ov::op::v0::Constant>(ov::element::f32,
+                                                      input_shapes[0].to_shape(),
+                                                      std::vector<float>{0.1122});
+    std::shared_ptr<Node> current = A;
+
+    for (size_t mm_count = 0; mm_count < num_input_nodes; ++mm_count) {
+        auto B = std::make_shared<ov::op::v0::Constant>(ov::element::f32, input_shapes[0].to_shape(), std::vector<float>{0.1122f + mm_count});
+        current = std::make_shared<ov::op::v0::MatMul>(current, B, false, true);
+        current = std::make_shared<ov::op::v1::Multiply>(current, add);
+        for (size_t i = 0; i < num_hidden_layers; ++i) {
+            auto constant = std::make_shared<ov::op::v0::Constant>(ov::element::f32,
+                                                                   ov::Shape{input_shapes[0].to_shape()[0]},
+                                                                   std::vector<float>{0.1122f + i});
+            current = std::make_shared<ov::op::v1::Add>(current, constant);
+        }
+    }
+    auto softmax = std::make_shared<ov::op::v8::Softmax>(current, 1);
+    auto result = std::make_shared<ov::op::v0::Result>(softmax);
+    return std::make_shared<Model>(ResultVector{result}, ParameterVector{A_param});
+}
+
+std::shared_ptr<ov::Model> MLPSeqFunction::initReference() const {
+    auto A_param = std::make_shared<ov::op::v0::Parameter>(precisions[0], input_shapes[0]);
+    std::shared_ptr<Node> A = A_param;
+    if (precisions[0] != ov::element::f32) {
+        A = std::make_shared<ov::op::v0::Convert>(A, ov::element::f32);
+    }
+
+    std::vector<std::shared_ptr<ov::Node>> constants;
+    for (size_t mm_count = 0; mm_count < num_input_nodes; ++mm_count) {
+        constants.push_back(std::make_shared<ov::op::v0::Constant>(
+            ov::element::f32, input_shapes[0].to_shape(), std::vector<float>{0.1122f + mm_count}));
+    }
+
+    std::vector<std::shared_ptr<ov::Node>> transposes;
+    for (size_t mm_count = 0; mm_count < num_input_nodes; ++mm_count) {
+        transposes.push_back(
+            std::make_shared<ov::op::v1::Transpose>(
+                constants[mm_count],
+                ov::op::v0::Constant::create(ov::element::i32, {input_shapes[0].get_shape().size()}, std::vector<int64_t>{1, 0})));
+    }
```

Comment:
Which pass insert Transpose ops? As far as I remember, `transposed_b=true` in `Brgemm` is handled by `BrgemmCopyB` and we don't need to insert `Transpose`

Reply:
Discussed offline and decided that this change is optional